### PR TITLE
[WIP] Server rewrite

### LINKED ISSUE #32: Rewrite server
Already being done on 'rewrite-server' branch..

### PR DESCRIPTION
**Pull request by ikornaselur**
Rewrote the back-end.

This can't be merged until all the basic functionality that's now on master is present after the rewrite.
### TODO
- [x] Serve the service worker for the browser
- [x] Allow the user to (un)subscribe to notification
- [x] Serve the manifest.json
- [x] Add the manifest.json to the main index file
- [x] Set up the uptime checks again

resolves #32

### CODE REVIEW

File: src/client/components/Status.js [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+          this.setState({
+            subscribed: true,
+            subDisabled: false,
+          });
+          return sendSubscriptionChange(subscription, 'subscribe');
+        })
+        .catch((e) => {
+          if (Notification.permission === 'denied') {
+            // The user denied
+            console.warn('Permission fro Notification was denied');
```
Comment (ikornaselur) [inline src/client/components/Status.js]: 🐔

File: src/client/components/Status.js [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+            this.setState({
+              subDisabled: false,
+              subscribed: false,
+            });
+          }
+        });
+    });
+  }
+
+  unsubscribe =() => {
```
Comment (ikornaselur) [inline src/client/components/Status.js]: 🐄

File: src/client/containers/CanaritusApp.js [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+export default class CanaritusApp extends Component {
+  static propTypes = {
+    error: PropTypes.any,
+    subscribed: PropTypes.bool,
+  };
+
+  render() {
+//     const actions = {
+//     };
```
Comment (ikornaselur) [inline src/client/containers/CanaritusApp.js]: 🐋

File: src/client/containers/CanaritusAppContainer.js [hunk 4]
```diff
@@ -0,0 +1,18 @@
+import {connect} from 'react-redux';
+import {bindActionCreators} from 'redux';
+
+import CanaritusApp from './CanaritusApp';
+import * as CanaritusActions from '../actions/CanaritusActions';
+
+const mapStateToProps = (state) => {
+  return {
+    error: state.error,
+    subscribed: state.subscribed,
```
Comment (ikornaselur) [inline src/client/containers/CanaritusAppContainer.js]: 🐖

File: src/client/reducers/index.js [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import {combineReducers} from 'redux';
+import statuses from './statuses';
+
+export default statuses;
```
Comment (ikornaselur) [inline src/client/reducers/index.js:4]: 🐓?

File: src/client/containers/CanaritusAppContainer.js (hunk 4 above)
Comment (ikornaselur) [inline src/client/containers/CanaritusAppContainer.js]: 🐘!

File: src/client/reducers/statuses.js [hunk 6]
```diff
@@ -0,0 +1,21 @@
+import {
+  GET_STATUSES,
+} from '../constants/ActionTypes';
+
+const initialState = {
+  error: null,
+  subscribed: false,
```
Comment (ikornaselur) [inline src/client/reducers/statuses.js]: 🐎

File: src/server/api/http.js [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+export const getStatus = (req, res) => {
+  const hosts = Object.keys(config.hosts);
+  res.json(hosts);
+};
+
+export const getHostStatus = (req, res) => {
+  const db = new Database('canaritus.db');
+  const host = req.params.host;
+
+  db.serialize(() => {
```
Comment (Ikornaselur) [inline src/server/api/http.js]: 🐡

File: src/server/api/http.js [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+};
+
+export const subscribe = (req, res) => {
+  const db = new Database('canaritus.db');
+  const id = req.body.id;
+  if (!id) {
+    res.status(400).send('id is missing from the post');
+  } else {
+    log('DB', `Adding ${id} to ids table`);
+    db.serialize(() => {
```
Comment (Ikornaselur) [inline src/server/api/http.js]: 🐡

**Pull request merged**